Cabal-syntax/src/Distribution/Fields/ParseResult.hs

```diff
@@ -73,6 +74,16 @@ runParseResult pr = unPR pr emptyPRState failure success
     -- If there are any errors, don't return the result
     success (PRState warns (err : errs) v) _ = (warns, Left (v, err :| errs))
 
+liftPR :: (a -> IO (ParseResult b)) -> ParseResult a -> IO (ParseResult b)
+liftPR f pr = unPR pr emptyPRState failure success
+  where
+    failure s = return $ PR $ \s' failure' _ -> failure' (concatPRState s s')
+    success s a = do
+      pr' <- f a
+      return $ PR $ \s' failure' success' -> unPR pr' (concatPRState s s') failure' success'
+    concatPRState (PRState warnings errors version) (PRState warnings' errors' version') =
+      (PRState (warnings ++ warnings') (toList errors ++ errors') (version <|> version'))
+
```

Comment:
@gbaz I migrated this from Legacy.hs:281 to the new ParseResult. I think you implemented the old version in Legacy.hs? Could you help reviewing this new version? As far as I see it works, but it would be awesome if you would take a look.